RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc

```diff
@@ -57,111 +167,84 @@ std::unique_ptr<MeasurementTracker> MeasurementTrackerESProducer::produce(const
-                                                    ptr_stripQuality,
-                                                    stripQualityFlags,
-                                                    stripQualityDebugFlags,
-                                                    ptr_pixelQuality,
-                                                    ptr_pixelCabling,
-                                                    pixelQualityFlags,
-                                                    pixelQualityDebugFlags,
-                                                    phase2TrackerCPE.product());
-  } else {
-    return std::make_unique<MeasurementTrackerImpl>(pset_,
-                                                    pixelCPE.product(),
-                                                    stripCPE.product(),
-                                                    hitMatcher.product(),
-                                                    trackerTopology.product(),
-                                                    trackerGeom.product(),
-                                                    geometricSearchTracker.product(),
-                                                    ptr_stripQuality,
-                                                    stripQualityFlags,
-                                                    stripQualityDebugFlags,
-                                                    ptr_pixelQuality,
-                                                    ptr_pixelCabling,
-                                                    pixelQualityFlags,
-                                                    pixelQualityDebugFlags);
+  desc.add<bool>("UseStripStripQualityDB", true)->setComment("read Strip status from SiStripQuality");
+  desc.addUntracked<bool>("DebugStripStripQualityDB", false)->setComment("dump out info on module status");
+
+  {
+    //used by MeasurementTrackerImpl
+    edm::ParameterSetDescription badStripCutsDesc;
+    badStripCutsDesc.add<uint32_t>("maxBad", 4);
+    badStripCutsDesc.add<uint32_t>("maxConsecutiveBad", 2);
+
+    edm::ParameterSetDescription mtiDesc;
+    mtiDesc.add<edm::ParameterSetDescription>("TIB", badStripCutsDesc);
+    mtiDesc.add<edm::ParameterSetDescription>("TOB", badStripCutsDesc);
+    mtiDesc.add<edm::ParameterSetDescription>("TID", badStripCutsDesc);
+    mtiDesc.add<edm::ParameterSetDescription>("TEC", badStripCutsDesc);
+
+    desc.addOptional<edm::ParameterSetDescription>("badStripCuts", mtiDesc);
   }
+
+  desc.add<bool>("UsePixelModuleQualityDB", true)->setComment("Use DB info at the module level (that is, detid level)");
+  desc.addUntracked<bool>("DebugPixelModuleQualityDB", false)->setComment("dump out info on module status");
+  desc.add<bool>("UsePixelROCQualityDB", true)->setComment("Use DB info at the ROC level");
+  desc.addUntracked<bool>("DebugPixelROCQualityDB", false)->setComment("dump out info om module status ");
+
+  descriptions.addDefault(desc);
```

Comment:
what is the reason there is no named description to replace MeasurementTrackerESProducer_cfi ?

Reply:
As the Core is adding `consumes` to all ESProducers, we've tried to minimize the possible impact of our changes (although sometimes we just can't help ourselves and feel compelled to re-write more code then necessary :) ). So I didn't want to touch the cfi.py files actually loaded in production. Plus MeasurementTrackerESProducer_cfi.py uses an `Era` which can not be done by the `fillDescriptions`.

Reply:
I worry that this is opening up a problem of large and complex duplication of the configuration with different implementation, which is bound to quickly diverge and become confusing.

Reply:
I can remove all the default values if you prefer.

Reply:
... and confuse HLT parsing as well.

One could replace courrent _cfi with a _cff one, which imports the auto-generated one and can still implemet the Era's. If so, one should also propagate whenever there is an inclusion of MeasurementTrackerESProducer_cfi.py  with the new MeasurementTrackerESProducer_cff.py

Reply:
Or even let the fillDescriptions define a MeasurementTrackerESProducerDefault_cfi (or similar)

Reply:
I have been considering having the `addDefault` actually allow generation of a `classtype_default.py` file which would declare a variable with name `_classtype_default`. The `_` guarantees that an `import *` statement or doing `process.load` would not pick it up accidentally but it still could be used for cloning.

Reply:
thinking of some other ways we were getting to the era-modified cases, this pattern seems reasonable to me (a generated cfi has a clearly distinct, not intuitively usable general name, the manually created cfi or cff would load it and clone to the usable name and also have the modifiers applied as necessary)

Reply:
So what do you want me to do?
1. leave as is
1. remove all default values in fillDescriptions except "Phase2StripCPE" which was previously using the `existsAs` trick to create a default value for a tracked parameter (which we do NOT want people to do)
1. have the fillDescriptions write a `MeasurementTrackerESProducerDefault_cfi.py` file and then change `MeasurementTrackerESProducer_cfi.py` to load that file in such a way as to not create 2 modules

Reply:
I do not want to have \~30 parameters (with somewhat complicated nested structure) defined in two places with some expectations of identity.
[the limit where it still seems acceptable to me is around O(1) parameter]

The proposal to generate ```classtype_default.py``` (perhaps still a \*cfi.py) looks good to me (perhaps I was not clear in https://github.com/cms-sw/cmssw/pull/27589#discussion_r307390394).
Please proceed.
Thank you.

Reply:
What I had proposed for `classtype_default.py` was a possible future extension to the parameter set system and would not be part of these pull requests. The 'option 3' (make a `MeasurementTrackerESProducerDefault_cfi.py` file using fillDescriptions) approximates that behavior based on what is possible now.

I will do 'option 3'.